### PR TITLE
refactor: enhance Alert and Toast components

### PR DESCRIPTION
**Pull request by cheton**
This PR primarily focues on the funciontality. I will create another PR to change severity colors if the visual spec is still uncertain in this moment.

### Todos
- [x] Add `redacted` fonts, a simple font alternative to the Lorem Ipsum text,
- [x] Severity levels
- [x] Add more icons
- [x] Enhance documentation for the `useToast` hook
- [x] ~`Closeable` & `CloseButton` components~
- [x] ~`useCloseable` Hook~
- [x] ~Wrap children with `Closeable` component~
- [x] Add `isCloseable` and `onClose` props to `Alert`, `AlertToast`, `Toast` components

### Recommended layouts for a toast notification

* Increase the vertical padding space to `4x` (or `1rem`) for a multiline paragraph.
* Set the minimum horizontal margin space to `4x` (or `1rem`) before the close button.
* Set the minimum horizontal margin space to `4x` (or `1rem`) between the side icon and the content.
* Apply vertical margin space with `2x` (or `.5rem`) between title and its content.
* Apply vertical margin space with `6x` (or `1.5rem`) between the end of the content and the action button (or action link).

#### `Alert` (offers severity levels)
https://trendmicro-frontend.github.io/styled-ui-demo/pr-112/alert

#### `Toast`
https://trendmicro-frontend.github.io/styled-ui-demo/pr-112/toast

#### `AlertToast` (offers severity levels)
https://trendmicro-frontend.github.io/styled-ui-demo/pr-112/alerttoast

#### `useToast` Hook
https://trendmicro-frontend.github.io/styled-ui-demo/pr-112/usetoast

### CODE REVIEW

**Comment by cheton** [comment]
Added in https://github.com/trendmicro-frontend/styled-ui/pull/112/commits/3a9be17a8d00573cc3d1d19d9232912ec175c879

## Toast

目前 `severity` prop 是跟 `Toast` 綁在一起，我有想把 Toast 拆成 `Toast` 和 `AlertToast` 這兩個 components，這樣使用起來會比較明確

#### `Toast` is a simple toast without severity

```jsx
<Toast />
```

image

#### `AlertToast` is a toast plus a severity level (`success`, `info`, `warning`, `error`)

```jsx
<AlertToast severity="success" />
<AlertToast severity="success" icon="_core.severity-success" />
<AlertToast severity="success" icon={<TMIcon name="close-s" />} />
```

image

File: packages/react-styled-ui/src/Closeable/index.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import { CloseableContext } from './context';
+
+const getMemoizedState = memoize(state => ({ ...state }));
+
+const Closeable = ({
+  isCloseable,
+  onClose,
+  children,
+}) => {
+  const onCloseCallback = useCallback(() => {
```
Comment (tinaClin) [inline packages/react-styled-ui/src/Closeable/index.js]: 看不懂useCloseable的用意....QQ

Comment (cheton) [reply packages/react-styled-ui/src/Closeable/index.js]: @tinaClin 某些方面的確是怪怪的，我就砍了先

Removed in d15b8e4d59ca595ad8553d2982660e799dd2f8be

**Review by roth1002 — APPROVED** [review]
👍 💯

**Comment by cheton** [comment]
TODO: 在 Toast 那頁加個提示有 AlertToast component

**Review by tinaClin — APPROVED** [review]

**Review by austinhung81 — APPROVED** [review]

**Pull request merged**